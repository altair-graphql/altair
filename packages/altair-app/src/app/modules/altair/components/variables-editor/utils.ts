import {
  getNullableType,
  GraphQLInputObjectType,
  GraphQLInputType,
  isEnumType,
  isLeafType,
  isListType,
  isNullableType,
  isSpecifiedScalarType,
} from 'graphql';
import {
  JSONSchema7,
  JSONSchema7Definition,
  JSONSchema7Type,
  JSONSchema7TypeName,
} from 'json-schema';

/**
 * Maximum depth to traverse the GraphQL type tree for variable type resolution.
 * This is to prevent slow performance because of resolving the variable types and/or infinite recursion.
 */
const MAX_DEPTH = 4;

export type VariableToType = Record<string, GraphQLInputType>;

const specifiedScalarTypeToJSONSchemaType: Record<string, JSONSchema7TypeName> =
  {
    String: 'string',
    Int: 'integer',
    Float: 'number',
    Boolean: 'boolean',
    ID: 'string',
  };
const wrapWithAnyOfNull = (def: JSONSchema7, wrap?: boolean) => {
  if (!wrap) {
    return def;
  }
  return {
    anyOf: [def, { type: 'null' as JSONSchema7TypeName }],
  };
};
export const graphqlInputTypeToJsonSchemaType = (
  originalType: GraphQLInputType,
  depth = 0,
  defaultValue?: JSONSchema7Type
): JSONSchema7 => {
  if (depth > MAX_DEPTH) {
    // Max depth reached. Default to string
    return { type: 'string' };
  }

  const type = getNullableType(originalType);

  // Handle enums
  if (isEnumType(type)) {
    return wrapWithAnyOfNull(
      {
        type: 'string',
        description: type.description ?? undefined,
        default: defaultValue,
        anyOf: type.getValues().map((v) => ({
          enum: [v.value],
          description: v.description ?? undefined,
        })),
      },
      isNullableType(originalType)
    );
  }
  if (isLeafType(type)) {
    if (isSpecifiedScalarType(type)) {
      return wrapWithAnyOfNull(
        {
          type:
            specifiedScalarTypeToJSONSchemaType[type.toString()] ?? 'string',
          description: type.description ?? undefined,
          default: defaultValue,
        },
        isNullableType(originalType)
      );
    }
    // custom scalar types default to string
    return wrapWithAnyOfNull(
      {
        type: 'string',
        description: type.description ?? undefined,
      },
      isNullableType(originalType)
    );
  }
  // Handle lists
  if (isListType(type)) {
    return wrapWithAnyOfNull(
      {
        type: 'array',
        items: graphqlInputTypeToJsonSchemaType(type.ofType, depth + 1),
        description: type.ofType?.description ?? undefined,
        default: defaultValue,
      },
      isNullableType(originalType)
    );
  }

  if (type instanceof GraphQLInputObjectType) {
    return {
      type: 'object',
      description: type.description ?? undefined,
      properties: Object.entries(type.getFields()).reduce(
        (acc, [key, field]) => {
          acc[key] = graphqlInputTypeToJsonSchemaType(
            field.type,
            depth + 1,
            field.defaultValue
          );
          return acc;
        },
        {} as Record<string, JSONSchema7Definition>
      ),
    };
  }

  // Unknown case. Default to string
  return { type: 'string' };
};

/*
 * Converts a VariableToType object to a JSON Schema object.
 * This is used to generate a JSON Schema for the variables editor.
 */
export const vttToJsonSchema = (vtt?: VariableToType): JSONSchema7 => {
  if (!vtt) {
    return {
      type: 'object',
      properties: {},
    };
  }
  return {
    type: 'object',
    properties: Object.entries(vtt).reduce((acc, [key, type]) => {
      acc[key] = graphqlInputTypeToJsonSchemaType(type);
      return acc;
    }, {} as Record<string, JSONSchema7Definition>),
  };
};
