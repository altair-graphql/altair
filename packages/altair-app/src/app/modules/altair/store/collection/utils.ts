import {
  CollectionState,
  IQueryCollection,
} from 'altair-graphql-core/build/types/state/collection.interfaces';
import { PerWindowState } from 'altair-graphql-core/build/types/state/per-window.interfaces';

export const COLLECTION_PATH_SEPARATOR = '/';

export const getCollections = (state: CollectionState) => state.list || [];
export const getCollection = (
  collections: IQueryCollection[],
  collectionId: string
) => collections.find((collection) => collection.id === collectionId);

export const getParentCollectionId = (collection: IQueryCollection) => {
  const id = collection.parentPath?.split(COLLECTION_PATH_SEPARATOR)?.pop();
  return id ? id : undefined;
};

export const getParentCollection = (
  collections: IQueryCollection[],
  collection: IQueryCollection
) => {
  const parentCollectionId = getParentCollectionId(collection);
  if (parentCollectionId) {
    return getCollection(collections, parentCollectionId);
  }
};

export const getAllParentCollections = (
  collections: IQueryCollection[],
  collection: IQueryCollection
) => {
  const result: IQueryCollection[] = [];
  let curCollection = collection;
  for (;;) {
    const parentCollection = getParentCollection(collections, curCollection);
    if (!parentCollection) {
      return result.reverse();
    }

    result.push(parentCollection);
    curCollection = parentCollection;
  }
};

export const getWindowParentCollections = (
  window: PerWindowState,
  collections: IQueryCollection[]
) => {
  if (window?.layout.windowIdInCollection && window?.layout.collectionId) {
    const collection = getCollection(collections, window.layout.collectionId);
    if (collection) {
      return [...getAllParentCollections(collections, collection), collection];
    }
  }
  return [];
};
