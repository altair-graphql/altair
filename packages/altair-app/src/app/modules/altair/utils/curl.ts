import { mapToKeyValueList } from '.';
import { IDictionary } from '../interfaces/shared';

export const parseCurlToObj = async (...args: any[]) =>
  (await import('curlup')).parseCurl(...args);

interface GenerateCurlOpts {
  url: string;
  method?: 'POST' | 'GET' | 'PUT' | 'DELETE';
  headers?: object;
  data?: { [key: string]: string };
}

const getCurlHeaderString = (header: { key: string; value: string }) => {
  return `-H '${header.key}: ${header.value}'`;
};

const buildUrl = (url: string, params?: { [key: string]: string }) => {
  const euc = encodeURIComponent;
  if (params) {
    const queryParams = Object.entries(params)
      .map(
        ([key, param]) =>
          euc(key) +
          '=' +
          euc(typeof param === 'string' ? param : JSON.stringify(param))
      )
      .join('&');

    return url.includes('?')
      ? `${url}&${queryParams}`
      : `${url}?${queryParams}`;
  }

  return url;
};

export const generateCurl = (opts: GenerateCurlOpts) => {
  const defaultHeaders = {
    'Accept-Encoding': 'gzip, deflate, br',
    'Content-Type': 'application/json',
    Accept: 'application/json',
    Connection: 'keep-alive',
    Origin: location.origin,
  };

  const method = opts.method || 'POST';

  const headers: IDictionary<string> = { ...defaultHeaders, ...opts.headers };
  const headerString = mapToKeyValueList(headers)
    .map(getCurlHeaderString)
    .join(' ');

  const url = method === 'GET' ? buildUrl(opts.url, opts.data) : opts.url;

  const curlParts = [`'${url}'`];

  if (!['GET', 'POST'].includes(method)) {
    curlParts.push(`-X ${method}`);
  }

  curlParts.push(`${headerString}`);

  if (method !== 'GET') {
    const dataBinary = `--data-binary '${JSON.stringify(opts.data)}'`;
    curlParts.push(dataBinary);
  }

  return `curl ${curlParts.join(' ')} --compressed`;
};
