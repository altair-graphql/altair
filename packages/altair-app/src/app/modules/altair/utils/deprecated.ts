const DEPRECATED_WINDOW_CONFIG = [
  '__ALTAIR_ENDPOINT_URL__',
  '__ALTAIR_SUBSCRIPTIONS_ENDPOINT__',
  '__ALTAIR_INITIAL_QUERY__',
  '__ALTAIR_INITIAL_VARIABLES__',
  '__ALTAIR_INITIAL_HEADERS__',
  '__ALTAIR_INITIAL_PRE_REQUEST_SCRIPT__',
  '__ALTAIR_INSTANCE_STORAGE_NAMESPACE__',
];

const logDeprecatedMessage = (key: string) => {
  console.warn(
    `DEPRECATION NOTICE: Configuring altair using global variables is deprecated, and will be removed in a future version.`
  );
  console.warn(`You set [window.${key}], which is deprecated.`);
  console.warn(`Use 'AltairGraphQL.init(opts)' instead.`);
};

export const handleDeprecations = () => {
  DEPRECATED_WINDOW_CONFIG.forEach((key) => {
    if ((window as any)[key]) {
      logDeprecatedMessage(key);
    }

    Object.defineProperty(window, key, {
      get() {
        return (window as any)[key];
      },
      set(value) {
        logDeprecatedMessage(key);
        (window as any)[key] = value;
      },
    });
  });
};
