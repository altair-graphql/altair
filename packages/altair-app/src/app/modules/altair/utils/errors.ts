import { EMPTY } from 'rxjs';
import { catchError } from 'rxjs/operators';
import { debug } from './logger';

export const catchUselessObservableError = catchError((err) => {
  debug.error(
    'Caught a supossedly useless observable error. Feel free to report this if this is an issue.'
  );
  debug.error(err);
  return EMPTY;
});

class BaseError extends Error {
  constructor(message: string, options?: { cause: unknown }) {
    super(message, options);
    this.name = this.constructor.name;
  }
}

export class InternalEditorError extends BaseError {
  constructor(cause: unknown) {
    super('Internal editor error occurred', { cause });
  }
}

export const getErrorResponse = async (err: unknown) => {
  if (err && typeof err === 'object') {
    // ky HTTP error: https://github.com/sindresorhus/ky#httperror
    if ('name' in err && err.name === 'HTTPError' && 'response' in err) {
      return (err.response as Response).json();
    }
  }

  return err;
};

export const getApiErrorCode = (err: unknown) => {
  if (!err) {
    return;
  }
  if (err && typeof err === 'object' && 'error' in err) {
    if (
      err.error &&
      typeof err.error === 'object' &&
      'code' in err.error &&
      typeof err.error.code === 'string'
    ) {
      return err.error.code;
    }
  }
};
