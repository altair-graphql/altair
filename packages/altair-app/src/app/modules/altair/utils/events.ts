/**
 * Event delegation
 * @param eventName
 * @param elSelector
 * @param fn
 * @example on('click', '.el', fn);
 */
export const on = (
  eventName: string,
  elSelector: string,
  fn: (e: Event) => void
) => {
  document.body.addEventListener(eventName, function (e: Event) {
    if (e.target && (e.target as Element).matches(elSelector)) {
      fn.apply(e.target, [e]);
    }
  });
};

export const handleExternalLinks = () => {
  on('click', 'a', function (e: Event) {
    const url = this.href || '';

    if (!url.replace(/#.*$/, '')) {
      // Handle hashes
      return;
    }
    e.preventDefault();

    // If electron app
    if (
      (window as any).process &&
      (window as any).process.versions['electron']
    ) {
      const electron = (window as any).require('electron');
      electron.shell.openExternal(url);
    } else {
      const win = window.open(url, '_blank');
      if (win) {
        win.focus();
      }
    }
  });
};
