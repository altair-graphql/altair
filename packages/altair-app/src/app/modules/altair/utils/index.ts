const toSnakeCase = require('to-snake-case'); // TODO: Check that this still works
import FileSaver from 'file-saver';
const commentRegex = require('comment-regex');
const validUrl = require('valid-url');
import isElectron from 'altair-graphql-core/build/utils/is_electron';
import { debug } from './logger';
import { IDictionary } from '../interfaces/shared';
import fileDialog from 'file-dialog';
import { VARIABLE_REGEX } from '../services/environment/environment.service';

/**
 * Download the specified data with the provided options
 * @param data data string to download
 * @param fileName name of downloaded file
 * @param opts configuration options
 */
export const downloadData = (
  data: string,
  fileName = 'data',
  {
    mimeType = 'text/plain',
    dataUriAttr = 'text/plain;charset=utf-8',
    fileType = 'txt'
  } = {}) => {

  const dataStr = `data:${dataUriAttr},${data}`;
  const fileNameWithExt = `${toSnakeCase(fileName)}.${fileType}`;
  const fileBlob = new Blob([data], {type: dataUriAttr});
  FileSaver.saveAs(fileBlob, fileNameWithExt);
};

/**
 * Download an object as a JSON file
 * @param obj The object to be downloaded
 * @param fileName The name the file will be called
 */
export const downloadJson = (obj: any, fileName = 'response', opts: any = undefined) => {
  let _opts = {
    mimeType: 'text/json',
    dataUriAttr: 'text/json;charset=utf-8',
    fileType: 'json'
  };

  if (opts) {
    _opts = { ..._opts, ...opts };
  }

  const dataStr = JSON.stringify(obj);
  downloadData(dataStr, fileName, _opts);
};

/**
 * Get file data as string
 * @param files FileList object
 */
export const getFileStr = (files: FileList) => {
  return new Promise<string>((resolve, reject) => {
    const fileReader = new FileReader();
    fileReader.onload = function (e: any) {
      const contents: string = e.target.result;

      // Resolve file content
      resolve(contents);
    };
    fileReader.readAsText(files[0]);
  });
};
interface FileDialogOptions {
  readonly multiple?: boolean;
  readonly accept?: string|ReadonlyArray<string>;
}
export const openFile = async (opts: FileDialogOptions = {}) => {
  try {
    const files = await fileDialog(opts);
    return getFileStr(files);
  } catch (err) {
    debug.log('There was an issue while opening the file: ', err);
  }
};

export const getFileContent = async (file: File) => {
  return new Promise<string>((resolve, reject) => {
    const fileReader = new FileReader();
    fileReader.onload = function (e: any) {
      const contents: string = e.target.result;

      // Resolve file content
      resolve(contents);
    };
    fileReader.readAsText(file);
  })
}

export const openFiles = async (opts: FileDialogOptions = {}) => {
  try {
    const files = await fileDialog({ ...opts, multiple: true });

    return Promise.all([...files].map((file) => getFileContent(file)));
  } catch (err) {
    debug.log('There was an issue while opening the files: ', err);

    return Promise.all([]);
  }
};

export const isExtension = !!((window as any).chrome && (window as any).chrome.runtime && (window as any).chrome.runtime.id);
export const isFirefoxExtension = !!((window as any).chrome && (window as any).chrome.geckoProfiler);

export const detectEnvironment = () => {
  if (isElectron) {
    return 'electron';
  }

  if (isExtension) {
    if (isFirefoxExtension) {
      return 'firefox-extension';
    } else {
      return 'chrome-extension'
    }
  }

  if (/http/.test(location.protocol)) {
    return 'web-app';
  }

  return 'other';
};

/**
 * Parse JSON with comments
 * @param str
 */
export const jsonc = (str: string) => {
  str = str.trim();
  str = str.replace(commentRegex(), '');

  if (!str) {
    return {};
  }
  return JSON.parse(str);
};

export const copyToClipboard = (str: string) => {
  const el = document.createElement('textarea');  // Create a <textarea> element
  el.value = str;                                 // Set its value to the string that you want copied
  el.setAttribute('readonly', '');                // Make it readonly to be tamper-proof
  el.style.position = 'absolute';
  el.style.left = '-9999px';                      // Move outside the screen to make it invisible
  document.body.appendChild(el);                  // Append the <textarea> element to the HTML document
  const documentSelection = document.getSelection();
  const selected =
    documentSelection && documentSelection.rangeCount > 0        // Check if there is any content selected previously
      ? documentSelection.getRangeAt(0)     // Store selection if found
      : false;                                    // Mark as false to know no selection existed before
  el.select();                                    // Select the <textarea> content
  document.execCommand('copy');                   // Copy - only works as a result of a user action (e.g. click events)
  document.body.removeChild(el);                  // Remove the <textarea> element
  if (selected && documentSelection) {                                 // If a selection existed before copying
    documentSelection.removeAllRanges();    // Unselect everything on the HTML document
    documentSelection.addRange(selected);   // Restore the original selection
  }
};

export const getFullUrl = (url: string, protocol = location.protocol) => {
  if (!url) {
    return url;
  }

  // regex to test if given string is an environment variable
  if (VARIABLE_REGEX.test(url)) {
    return url;
  }

  if (!validUrl.isUri(url)) {
    if (url.trim() === '*') {
      return location.href;
    }

    if (url.substr(0, 1) === '/') {
      url = url.substr(1);
    }
    return `${protocol.replace(/:$/, '').toLowerCase()}://${location.host}/${url}`;
  }

  return url;
};

export function parseDotNotationKey(key: string) {
  const intKey = parseInt(key, 10);
  return intKey.toString() === key
    ? intKey
    : key;
}

/**
 * Sets value to object using dot notation.
 * @param obj Object to set the value to.
 * @param path Path as a string, separated by dots.
 * @param value Any value to be set.
 * @example
 * ```ts
 * const obj = { a: 1, b: { c: 2 } };
 * setByDotNotation(obj, "b.d.e.0", 3);
 * // results in
 * { a: 1, b: { c: 2, d: { e: [3] } } };
 * ```
 */
export function setByDotNotation<TResult = any>(
  obj: IDictionary,
  path: Array<number | string> | number | string,
  value: TResult,
): TResult | undefined {
  if (typeof path === 'number') {
    path = [path];
  }
  if (!path || path.length === 0) {
    return undefined;
  }
  if (typeof path === 'string') {
    return setByDotNotation(obj, path.split('.').map(parseDotNotationKey), value);
  }

  const currentPath = path[0];
  const currentValue = obj[currentPath];

  if (path.length === 1) {
    obj[currentPath] = value;
    return currentValue;
  }

  if (currentValue === undefined) {
    if (typeof path[1] === 'number') {
      obj[currentPath] = [];
    } else {
      obj[currentPath] = {};
    }
  }

  return setByDotNotation(obj[currentPath], path.slice(1), value);
}

export const mapToKeyValueList = (obj: Record<string, string>) => {
  return Object.keys(obj).filter(key => !!key).map(key => ({ key, value: obj[key] }))
};

export function truncateText(text: string, maxLength = 70) {
  let appendEllipsis = false;
  if (text.length > maxLength) {
    appendEllipsis = true;
  }

  return text.substring(0, maxLength) + (appendEllipsis ? '...' : '');
}

export const externalLink = (e: Event, url: string) => {
  e.preventDefault();

  // If electron app
  if ((window as any).process && (window as any).process.versions.electron) {
    const electron = (window as any).require('electron');
    electron.shell.openExternal(url);
  } else {
    const win = window.open(url, '_blank');
    if (win) {
      win.focus();
    }
  }
}