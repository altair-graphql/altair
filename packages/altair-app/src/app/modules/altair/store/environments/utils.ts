import {
  ENVIRONMENT_VARIABLE_SOURCE_TYPE,
  EnvironmentsState,
  EnvironmentVariables,
  EnvironmentVariableSourceType,
  IEnvironment,
} from 'altair-graphql-core/build/types/state/environments.interfaces';
import { merge } from 'lodash-es';
import { parseJson } from '../../utils';
import { IQueryCollection } from 'altair-graphql-core/build/types/state/collection.interfaces';

export const mergeEnvironments = (env1: IEnvironment, env2: IEnvironment) => {
  return merge(env1, env2);
};

type EnvironmentWithMetadata = {
  environment: IEnvironment;
  sourceType: EnvironmentVariableSourceType;
  sourceName: string;
};
/**
 * Get active environments. Most significant environment is last in the list.
 */
export const getActiveEnvironmentsList = (
  environmentsState: EnvironmentsState,
  windowCollections: IQueryCollection[] = []
): EnvironmentWithMetadata[] => {
  const baseEnvironment = parseJson(environmentsState.base.variablesJson, {});
  let subEnvironment: IEnvironment | undefined;

  if (environmentsState.activeSubEnvironment) {
    const activeSubEnvironment = environmentsState.activeSubEnvironment;
    const activeSubEnvState = environmentsState.subEnvironments.find((env) => {
      return env.id === activeSubEnvironment;
    });

    if (activeSubEnvState) {
      subEnvironment = parseJson(activeSubEnvState.variablesJson, {});
    }
  }

  return [
    {
      environment: baseEnvironment,
      sourceType: ENVIRONMENT_VARIABLE_SOURCE_TYPE.BASE_ENVIRONMENT,
      sourceName: 'Global Environment',
    },
    ...(subEnvironment
      ? [
          {
            environment: subEnvironment,
            sourceType: ENVIRONMENT_VARIABLE_SOURCE_TYPE.SUB_ENVIRONMENT,
            sourceName: 'Environment',
          },
        ]
      : []),
    ...(windowCollections
      .map((collection) => {
        if (!collection.environmentVariables) {
          return;
        }
        return {
          environment: collection.environmentVariables,
          sourceType: ENVIRONMENT_VARIABLE_SOURCE_TYPE.COLLECTION,
          sourceName: collection.title || 'Collection',
        };
      })
      .filter((env) => !!env) as EnvironmentWithMetadata[]),
  ];
};

export const getActiveEnvironment = (
  environmentsState: EnvironmentsState,
  windowCollections: IQueryCollection[] = []
) => {
  const environmentsList = getActiveEnvironmentsList(
    environmentsState,
    windowCollections
  );
  return environmentsList.reduce((acc, cur) => {
    return mergeEnvironments(acc, cur.environment);
  }, {} as IEnvironment);
};

const flattenEnvironment = (
  result: EnvironmentVariables,
  obj: IEnvironment,
  sourceType: EnvironmentVariableSourceType,
  sourceName: string,
  prefix = ''
) => {
  for (const [key, value] of Object.entries(obj)) {
    const fullKey = prefix ? `${prefix}.${key}` : key;

    if (value !== null && typeof value === 'object' && !Array.isArray(value)) {
      // Recursively flatten nested objects
      flattenEnvironment(
        result,
        value as IEnvironment,
        sourceType,
        sourceName,
        fullKey
      );
    } else {
      // Store the variable data
      result[fullKey] = {
        key: fullKey,
        value,
        sourceType,
        sourceName,
      };
    }
  }
};
export const environmentsToEnvironmentVariables = (
  environments: EnvironmentWithMetadata[]
): EnvironmentVariables => {
  const result: EnvironmentVariables = {};

  // Process each environment in order (later ones override earlier ones)
  for (const { environment, sourceType, sourceName } of environments) {
    flattenEnvironment(result, environment, sourceType, sourceName);
  }

  return result;
};
