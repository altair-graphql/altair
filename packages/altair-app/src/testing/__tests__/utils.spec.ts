import { describe, it, expect } from '@jest/globals';
import { buildTestHostComponentTemplate } from '../utils';

describe('utils', () => {
  describe('buildTestHostComponentTemplate', () => {
    it('should return expected template string', () => {
      const tmpl = buildTestHostComponentTemplate(
        'my-main-component',
        ['disabled', 'input2', 'collections'],
        ['saveChange', 'dialogChange', 'modelChange']
      );

      expect(tmpl).toMatchSnapshot();
    });
  });
});
