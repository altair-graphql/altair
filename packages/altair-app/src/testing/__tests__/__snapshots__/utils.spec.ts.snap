// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`utils buildTestHostComponentTemplate should return expected template string 1`] = `
"
    <div class="test-host">
      <my-main-component
        [disabled]="inputs.disabled"
[input2]="inputs.input2"
[collections]="inputs.collections"

        (saveChange)="saveChange($event)"
(dialogChange)="dialogChange($event)"
(modelChange)="modelChange($event)"
      ></my-main-component>
    </div>
  "
`;
